subset/alliedSwitch/src/main/java/alliedswitch/AlliedTelnetClientSocket.java

```diff
@@ -0,0 +1,425 @@
+              if (position >= 0) {
+                interrogator.setUserEnabled(true);
+                expectedLength = 2;
+                charLength = interrogator.getHostname().length() + 1;
+                flush = 0;
+                beginPosition = 0;
+                count++;
+              }
+            } else if (interrogator.getRequestFlag() == 1 || interrogator.getRequestFlag() == 2) {
+              position = rxGathered.lastIndexOf(interrogator.getHostname());
+              if (position >= 0) {
+                expectedLength = 600;
+                charLength = interrogator.getHostname().length() + 1;
+                flush = 0;
+                if (rxGathered.length() >= expectedLength) {
+                  beginPosition = 4;
+                  count++;
+                }
+              }
+            } else if (interrogator.getRequestFlag() == 3) {
+              position = rxGathered.indexOf("end");
+              if (position >= 0) {
+                expectedLength = 1000;
+                charLength = 3;
+                flush = 15;
+                count++;
+              }
+            } else if (interrogator.getRequestFlag() == 4) {
+              position = rxGathered.lastIndexOf(interrogator.getHostname());
+              if (position >= 0) {
+                expectedLength = 290;
+                charLength = -1;
+                flush = 0;
+                if (rxGathered.length() >= expectedLength) {
+                  beginPosition = 4;
+                  count++;
+                }
+              }
+            }
+
+            if (position >= 0 && rxGathered.length() >= expectedLength) {
+              parseFlag = true;
+            }
+
+            if (parseFlag) {
+              rxGathered = rxGathered.substring(beginPosition, position + charLength);
+              System.out.println(
+                  java.time.LocalTime.now()
+                      + "rxGatheredLen:"
+                      + rxGathered.length()
+                      + "rxGathered:"
+                      + rxGathered);
+              rxData.delete(0, position + charLength + flush);
+
+              interrogator.receiveData(rxGathered);
+
+              parseFlag = false;
+            }
+          }
+        }
+      } catch (InterruptedException e) {
+        System.err.println("InterruptedException gatherData:" + e.getMessage());
+        e.printStackTrace();
+      }
+    }
+  }
+
+  public void writeData(String data) {
+    Runnable runnable =
+        () -> {
+          writeOutputStream(data);
+        };
+    Thread writeThread = new Thread(runnable);
+    writeThread.start();
+  }
+
+  private void writeOutputStream(String data) {
+    try {
+      outputStream.write(data.getBytes());
+      // outputStream.write(data.getBytes(StandardCharsets.UTF_8));
+      outputStream.flush();
+    } catch (IOException e) {
+      System.err.println("Exception while writing socket:" + e.getMessage());
+      e.printStackTrace();
```

Comment:
if you have e.printStackTrace() you don't need the System.err

Reply:
thanks for the tip :)